cli/src/pcluster/models/cluster_slurm.py

```diff
@@ -104,3 +107,19 @@ def __init__(
     ):
         super().__init__(image, head_node, shared_storage, monitoring, tags, iam, custom_actions)
         self.scheduling = scheduling
+
+        for queue in self.scheduling.queues:
+            for compute_resource in queue.compute_resources:
+                self._add_validator(
+                    InstanceArchitectureCompatibilityValidator,
+                    instance_type=compute_resource.instance_type,
+                    architecture=self.architecture,
+                )
+                if compute_resource.efa:
+                    self._add_validator(
+                        EfaOsArchitectureValidator,
+                        priority=9,
+                        efa_enabled=compute_resource.efa.enabled,
+                        os=self.image.os,
+                        architecture=self.architecture,
+                    )
```

Comment:
Do we really want to replicate the code for Awsbatchcluster?

Reply:
Another good point. I wasn't sure either. 
The decision here is whether to define a generic interface for the ComputeResource and Scheduling classes.
For now let's go ahead and we can go back to this decision.